trace/opencensus/trace/ext/django/middleware.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Django middleware helper to capture and trace a request."""
+import logging
+import threading
+
+from opencensus.trace.ext import utils
+from opencensus.trace.ext.django.config import settings
+from opencensus.trace.propagation.google_cloud_format import from_header
+
+_thread_locals = threading.local()
+
+HTTP_METHOD = '/http/method'
+HTTP_URL = '/http/url'
+HTTP_STATUS_CODE = '/http/status_code'
+
+REQUEST_SPAN_NAME = 'django_request_span'
+TRACER_THREAD_LOCAL_KEY = '_opencensus_django_request_tracer'
+
+_DJANGO_TRACE_HEADER = 'HTTP_X_CLOUD_TRACE_CONTEXT'
+
+log = logging.getLogger(__name__)
+
+
+def _get_django_request():
+    """Get Django request from thread local.
+
+    :rtype: str
+    :returns: Django request.
+    """
+    return getattr(_thread_locals, 'request', None)
+
+def _get_current_request_tracer():
+    """Get the current request tracer."""
+    return getattr(_thread_locals, TRACER_THREAD_LOCAL_KEY, None)
+
+def _set_django_labels(span, request):
+    """Set the django related labels."""
+    django_user = getattr(request, 'user', None)
+
+    if django_user is None:
+        return
+
+    user_id = django_user.getattr(django_user, 'pk', None)
```

Comment:
I guess `pk` is the primary key of the `django.contrib.auth.models.User` object?  Maybe a comment or a link to documentation?